### ISSUE TITLE
Support for customized XMLs

### ISSUE THREAD

**Issue by Zintixx** [WeblateOrg/weblate#4158]
**Is your feature request related to a problem? Please describe.**
The current XML formats Weblate takes don't allow for flexibility in terms of how different XMLs are structured. I'm currently having the need to convert it to a flat XML for each individual file and once the translation is done, I need to convert it back. It becomes non-ideal for a smooth automated workflow.

**Describe the solution you'd like**
To give some context, here is an XML file I have:
https://github.com/Zintixx/MapleStory2-English/blob/master/Latest-KMS2/string/en/achievecategoryname.xml
In this example, it has two attribute I would like it to grab. The identifier/key here would be "id" and the string I would want to translate is "name"

In another example:
https://github.com/Zintixx/MapleStory2-English/blob/master/Latest-KMS2/string/en/achievedescription.xml
This has four strings per ID I would want it to translate (desc, complete, manualDesc, and manualComplete).

I do also want to point out these are all nested in a "ms2" tag rather than "root." 

The solution for this that I would want is to be able to feed Weblate these XMLs, read it, and ask which attribute would be the unique identifier/key and which attribute(s) to accept that needs to be translated.

**Describe alternatives you've considered**
A slight alternative to this would be rather than asking which attributes to accept that would need translation is to include all of the attributes (besides the unique identifier/key).

**Comment by nijel** [comment]
This can be split to two separate topics:

* Extend flat XML format to support attribute extraction (right now it extracts only tag content).
* Add support for configuring this inside Weblate. This has been already brought up at https://github.com/WeblateOrg/weblate/issues/1897, but it has been postponed back then.
